### PR TITLE
Fix a bug in CodeDomComponentSerializationService.CodeDomSerializationStore.ComponentListCodeDomSerializer

### LINKED ISSUE #10096: System.NullReferenceException: 'Object reference not set to an instance of an object.' occurs when delete button control then click undo in DemoConsole application
### .NET version

.Net 9.0: Main branch of Winforms repo

### Did it work in .NET Framework?

Not tested/verified

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, it works well in .Net 8.0 from Release/8.0 branch of Winforms repo. The original issue is [9412](https://github.com/dotnet/winforms/issues/9412).

### Issue description

In [DemoConsole](https://github.com/dotnet/winforms/tree/main/src/System.Windows.Forms/tests/IntegrationTests/DesignSurface/DemoConsole) test app of Winforms repo, System.NullReferenceException: 'Object reference not set to an instance of an object.' occurs when delete button control then click undo.
Actual result in DemoConsole application when **testing main branch**:

https://github.com/dotnet/winforms/assets/94418985/80c58b85-6df8-4d5b-acf7-2a7655cbc567

**Call Stack:**
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=System.Windows.Forms.Design
  StackTrace:
   at System.ComponentModel.Design.Serialization.CodeDomComponentSerializationService.CodeDomSerializationStore.ComponentListCodeDomSerializer.DeserializeModifier(IDesignerSerializationManager manager, String name, Object state) in C:\test\winforms\src\System.Windows.Forms.Design\src\System\ComponentModel\Design\Serialization\CodeDomComponentSerializationService.CodeDomSerializationStore.ComponentListCodeDomSerializer.cs:line 421

  This exception was originally thrown at this call stack:
    System.ComponentModel.Design.Serialization.CodeDomComponentSerializationService.CodeDomSerializationStore.ComponentListCodeDomSerializer.DeserializeModifier(System.ComponentModel.Design.Serialization.IDesignerSerializationManager, string, object) in CodeDomComponentSerializationService.CodeDomSerializationStore.ComponentListCodeDomSerializer.cs

Expected result in DemoConsole application when **testing release/8.0 branch**:

https://github.com/dotnet/winforms/assets/94418985/bcaf868f-1ff7-4f16-bbf5-942a3e1b1cfe

### Steps to reproduce

Test sample: [DemoConsole](https://github.com/dotnet/winforms/tree/main/src/System.Windows.Forms/tests/IntegrationTests/DesignSurface/DemoConsole) in Winforms repo

### PR DESCRIPTION
**Pull request by halgab**
The bug was introduced in #9120

Fixes #10096

## Test methodology 

- I ran DemoConsole and verified manually that the bug no longer reproduces

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10122)

### CODE REVIEW

**Review by LeafShi1 — APPROVED** [review]

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.ComponentListCodeDomSerializer.cs [hunk 1]
```diff
@@ -414,6 +414,11 @@ private static void DeserializeEventResets(IDesignerSerializationManager? manage
 
             private static void DeserializeModifier(IDesignerSerializationManager manager, string name, object? state)
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.ComponentListCodeDomSerializer.cs:415]: Maybe this shouldn't be marked as nullable?

Comment (halgab) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/CodeDomComponentSerializationService.CodeDomSerializationStore.ComponentListCodeDomSerializer.cs:415]: The check used to take place outside of the method. #9120 changed it to move the check inside of the method, for all methods called in `DeserializeState` (except this one were the job was half done). So it makes sense as the code is now written to have it nullable.

**Comment by halgab** [comment]
> Are we able to add unit test for this?

I'll see what I can do. I'm taking a look right now, and apparently we test `CodeDomComponentSerializationService` for serialisation, but not for deserialisation...

**Comment by lonitra** [comment]
> I'll see what I can do. I'm taking a look right now, and apparently we test `CodeDomComponentSerializationService` for serialisation, but not for deserialisation...
> 
> Edit: This doesn't look like I'll be able to add those easily. Trying to deserialise at the end of the serialisation tests make them crash, and investigating this will likely take some time (at least to me)

Thank you for taking a look! That's interesting that we don't have tests for deserialization. I don't want to block on this, but could you follow up by opening an issue on adding deserialization tests here.

**Review by lonitra — APPROVED** [review]
Thanks for jumping on this 🚀

**Comment by halgab** [comment]
> > I'll see what I can do. I'm taking a look right now, and apparently we test `CodeDomComponentSerializationService` for serialisation, but not for deserialisation...
> > Edit: This doesn't look like I'll be able to add those easily. Trying to deserialise at the end of the serialisation tests make them crash, and investigating this will likely take some time (at least to me)
> 
> Thank you for taking a look! That's interesting that we don't have tests for deserialization. I don't want to block on this, but could you follow up by opening an issue on adding deserialization tests here.

Sure. #719 seems to be related, but I opened #10135 anyway

**Pull request merged**